### PR TITLE
final remarks

### PR DESCRIPTION
**Pull request by dzheki4ak**

### CODE REVIEW

File: src/components/BalancePage/BalancePage.jsx [hunk 1]
```diff
@@ -5,14 +5,24 @@ import { useSelector } from 'react-redux';
...
+import Spinner from '../Spinner/Spinner.jsx';
+
+import './BalancePage.scss';
 
 const BalancePage = () => {
   const list = useSelector(transactionsTotalsSelector);
 
-  const { totalsCalculation, inputs, outputs, inputValue, outputValue } =
-    helpers;
+  if (list.length < 1) {
```
Comment (vitcool) [inline src/components/BalancePage/BalancePage.jsx:23]: OK, probably better will check if `list.length === 0`
I would suggest to create separate variable `const isListEmpty = list.length === 0` and then use it here ;)

File: src/components/TransactionDetails/TransactionDetails.jsx [hunk 2]
```diff
@@ -6,50 +6,56 @@ import { transactionsTotalsSelector } from '../../redux/modules/transactionsList
...
 
 import { HOME, TRANSACTIONS } from '../../routes';
 
 const TransactionDetails = ({ location }) => {
   const list = useSelector(transactionsTotalsSelector);
 
+  if (list.length < 1) {
+    return <Spinner />;
+  }
+
```
Comment (vitcool) [inline src/components/TransactionDetails/TransactionDetails.jsx:19]: oh, you use this checking three times (first one - here, second - BalancePage and third - in TransactionsListItem), I am thinking about the wrapper component with the following content:
```
const TransactionsWrapper = ({ children, list }) => {
const isListEmpty = list.length === 0

if (listIsEmpty) {
  return <Spinner />;
}

return children;
}
```

then use it here lnside return function
```
<TransactionsWrapper>
  <div className="transactions_list__page">
    ....
  </div>
</TransactionsWrapper>
```

File: src/components/TransactionDetails/TransactionDetailsList.jsx [hunk 3]
```diff
@@ -1,32 +1,12 @@
-/* eslint-disable camelcase */
 import React from 'react';
+import TransactionDetailsListItem from './TransactionDetailsListItem.jsx';
```
Comment (vitcool) [inline src/components/TransactionDetails/TransactionDetailsList.jsx:3]: could you add empty line between 3d party libraries from package.json and your own code imports please?

File: src/components/TransactionDetails/TransactionDetailsListItem.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable camelcase */
```
Comment (vitcool) [inline src/components/TransactionDetails/TransactionDetailsListItem.jsx]: could you explain why do you need it here please?

File: src/components/TransactionDetails/TransactionDetailsListItem.jsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import React from 'react';
+
+const TransactionDetailsListItem = ({ operations }) => (
+    operations.map(opr => {
+      const { output_value, value, spent_by } = opr;
+      const [address] = opr.addresses;
```
Comment (vitcool) [inline src/components/TransactionDetails/TransactionDetailsListItem.jsx:6]: nice solution ;)

File: src/redux/modules/transactionsList/reducer.js [hunk 6]
```diff
@@ -1,6 +1,8 @@
 import { FETCHED } from './actions.js';
 
-const initialState = {};
+const initialState = {
+  transactionsList: [],
```
Comment (vitcool) [inline src/redux/modules/transactionsList/reducer.js]: could you rename it to `data`, please?
looks like you have `transactionsList.transactionsList`

so you'll have
```
case FETCHED:
      return {
        ...state,
        data: action.payload,
      };
 ```

**Review by vitcool — COMMENTED** [review]
could you check my comments please?

File: src/components/TransactionDetails/TransactionDetailsListItem.jsx (hunk 4 above)
Comment (dzheki4ak) [reply src/components/TransactionDetails/TransactionDetailsListItem.jsx]: My es-lint barks that my variable is not in camelCase.

File: src/redux/modules/transactionsList/reducer.js (hunk 6 above)
Comment (dzheki4ak) [reply src/redux/modules/transactionsList/reducer.js]: DOne

**Pull request merged**